client/modules/core/main.js

```diff
@@ -661,6 +669,44 @@ export default {
     return settings.settings || {};
   },
 
+  getShopLanguage() {
+    const shopId = this.getShopId();
+    const shop = Shops.findOne({ _id: shopId });
+    return shop ? shop.language : "";
+  },
+
+  getSlug(slugString) {
+    const lazyLoadSlugify = async () => {
+      let mod;
+      const lang = this.getShopLanguage();
+
+      // If slugify has been loaded but language has changed to non latin-based language, load transliteration
+      if (slugify && slugify.name === "replace" && latinLangs.indexOf(lang) === -1) {
+        mod = await import("transliteration");
+      } else if (slugify) {
+        // If slugify/transliteration is loaded & no lang change
+        return;
+      } else if (latinLangs.indexOf(lang) >= 0) {
+        // If shop's language uses latin based chars, load slugify, else load transliterations's slugify
+        mod = await import("slugify");
+      } else {
+        mod = await import("transliteration");
+      }
+
+      // Slugify is exported to modules.default while transliteration is exported to modules.slugify
+      slugify = mod.default || mod.slugify;
+    };
+
+    let slug;
+    Promise.resolve(lazyLoadSlugify()); // eslint-disable-line promise/catch-or-return
```

Comment:
`Promise.resolve` doesn't do anything here. It won't await, so the first time this is run, the return will always be "".

You can see this issue by opening the console and entering `ReactionCore.getSlug("abc")` twice. It only works the second time.

I suspect this was probably an existing issue that we just hadn't noticed.

I think the only solution is to make this whole `getSlug` function async, and await `lazyLoadSlugify()` here.

The places I see that will then need to be changed:
- `getPrimaryShopPrefix` in the same file would need to be made async, but it looks to be unused. If so, just delete it.
- `getShopPrefix` in the same file would need to be made async, but it looks to be unused. If so, just delete it. (`Reaction.getShopPrefix` is called a few times, but only from server code as far as I can see.)
- Call to `Reaction.getSlug` in `/imports/plugins/core/ui-tagnav/client/helpers/tags.js` needs to be made async
- Call to `Reaction.getSlug` in `/imports/plugins/core/ui/client/containers/tagListContainer.js` needs to be made async (btw, this `updateSuggestions` function is identical to the previous one, so maybe move that to a util file and share it?)